### PR TITLE
[CICD] trigger docker build on specific branches by default

### PR DESCRIPTION
**Pull request by geekflyer**
to avoid AWS codebuild quota issues

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -285,6 +285,13 @@ workflows:
       - unit-test
       - docker-build-push:
           context: aws-dev
+          filters:
+            branches:
+              only:
+                - main
+                - auto
+                - canary
```
Comment (sherry-x) [inline .circleci/config.yml:293]: Can we add a testnet branch too? Thanks!

Comment (geekflyer) [reply .circleci/config.yml:293]: done

Comment (sherry-x) [reply .circleci/config.yml:293]: Let’s keep both of them, testnet and devnet are both in use now

**Review by sherry-x — APPROVED** [review]

**Pull request merged**